### PR TITLE
[[ Bug 19279 ]] [MobilePickMedia] Prevent crash and correct return value

### PR DESCRIPTION
**Pull request by HedgeHao**

### CODE REVIEW

File: engine/src/java/com/runrev/android/Engine.java [hunk 1]
```diff
@@ -3153,15 +3153,20 @@ private void onMediaResult(int resultCode, Intent data)
...
+				{
+					t_cursor = ((LiveCodeActivity) getContext()).getContentResolver().query(t_data, null, null, null, null);
+				}catch(SecurityException e){}
                 if (t_cursor != null)
                 {
                     t_cursor.moveToFirst();
                     int t_index = t_cursor.getColumnIndex (MediaStore.Images.ImageColumns.DATA);
-                    t_path = t_cursor.getString(t_index);
+                    if (t_index > 0)
+                    		t_path = t_cursor.getString(t_index);
```
Comment (peter-b) [inline engine/src/java/com/runrev/android/Engine.java]: Hi Hedgehao,

I think you can avoid this check by rearranging some code into the `try` block:

```java
Cursor t_cursor = null;
try
{
    t_cursor = ((LiveCodeActivity) getContext()).getContentResolver().query(t_data, null, null, null, null);
    if (t_cursor != null)
    {
        t_cursor.moveToFirst();
        int t_index = t_cursor.getColumnIndex (MediaStore.Images.ImageColumns.DATA);
        t_path = t_cursor.getString(t_index);
    }
} catch (SecurityException e) {}
```

**Comment by peter-b** [comment]
@HedgeHao I made a couple of minor tweaks.  Thanks for the patch!

@livecode-vulcan review by @peter-b ok ef0cb48

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok ef0cb48902b7c5c72bab0c047d20a24d0e505d91

**Comment by livecode-vulcan** [comment]
😎 test success ef0cb48902b7c5c72bab0c047d20a24d0e505d91

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/1027)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/1033)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/1025)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/1026)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/1033)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/1047)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/1024)

**Pull request merged**